src/averageAges.js

```diff
@@ -36,9 +48,25 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
-}
+  let women = people.filter(obj => {
+    return obj['sex'] === 'f';
+  });
+
+  if (withChildren) {
+    women = women.filter(mother => {
+      return people.some(child => mother.name === child.mother);
+    });
+  };
+
+  const totalAge = women.reduce((total, obj) => {
+    return total + (obj.died - obj.born);
+  }, 0);
+
+  return totalAge / women.length;
```

Comment:
You've already used these code twice. So, create a separated function to make these calculations